### DISCUSSION TITLE
Will `wsl --update` when using WSL 1 update it to the store version?

### DISCUSSION THREAD

**Discussion by Yay295** [microsoft/WSL#9399] [Q&A] [answered]
And if it does, will it still be WSL 1, or will it be WSL 2?

**Comment by Biswa96** [comment, marked as answer]
WSL1 does not use the WSL from Windows Store. Support for running WSL 1 distros still requires the “Windows Subsystem for Linux” optional component. See https://devblogs.microsoft.com/commandline/the-windows-subsystem-for-linux-in-the-microsoft-store-is-now-generally-available-on-windows-10-and-11/

**Reply by Yay295** [reply]
So what will `wsl --update` do if I'm using WSL 1?

**Reply by NotTheDr01ds** [reply]
> So what will `wsl --update` do if I'm using WSL 1?

It should be a no-op.  Prior to the Store version of WSL2, the purpose of `wsl --update` was to update the WSL2 *kernel*.  Since there's no kernel package in WSL1, I can't imagine it is going to do anything.

**Reply by Yay295** [reply]
Okay, thanks.
